test/assert.t.cpp

```diff
@@ -36,6 +36,9 @@ enum Color
 #endif
 { red, green, blue };
 
+#if gsl_COMPILER_NVHPC_VERSION
+# pragma diag_suppress 941 // Suppress:  missing return statement at end of non-void function "..."
```

Comment:
The entire point of this test case is to prove that the code compiles warning-free if `gsl_FailFast()` is used, so I'd suppress it here only as a last resort. Does the `__attribute__((noreturn))` annotation in `fail_fast_terminate()` and `fail_fast_throw()` not suffice to suppress the warning?

Reply:
Unfortunately it doesn't seem to. I also tried with an explicit NVHPC clause in
```
#elif gsl_COMPILER_GNUC_VERSION || gsl_COMPILER_CLANG_VERSION || gsl_COMPILER_APPLECLANG_VERSION || gsl_COMPILER_ARMCC_VERSION || gsl_COMPILER_NVHPC_VERSION
# define gsl_NORETURN  __attribute__((noreturn))
#else
```

I obviously agree the suppression isn't a nice solution, I just thought I'd push the result of testing your suggestions. The current head of the PR builds without warnings for me, and the tests pass.

Reply:
Oh well. Thanks for checking.